docs/metrics.md

```diff
@@ -0,0 +1,561 @@
+    section from the supplementary guidelines for more information.
+
+| OpenTelemetry Specification | OTel Rust Instrument Type |
+| --------------------------- | -------------------- |
+| [Asynchronous Counter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-counter) | [`ObservableCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableCounter.html) |
+| [Asynchronous Gauge](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-gauge) | [`ObservableGauge`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableGauge.html) |
+| [Asynchronous UpDownCounter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-updowncounter) | [`ObservableUpDownCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.ObservableUpDownCounter.html) |
+| [Counter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#counter) | [`Counter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Counter.html) |
+| [Gauge](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#gauge) | [`Gauge`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Gauge.html) |
+| [Histogram](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#histogram) | [`Histogram`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.Histogram.html) |
+| [UpDownCounter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#updowncounter) | [`UpDownCounter`](https://docs.rs/opentelemetry/latest/opentelemetry/metrics/struct.UpDownCounter.html) |
+
+:stop_sign: You should avoid creating instruments (e.g. `Counter`) too
+frequently. Instruments are fairly expensive and meant to be reused. For most
+applications, instruments can be created once and re-used. Instruments can also
+be cloned to create multiple handles to the same instrument, but the cloning
+should not be on hot path, but instead the cloned instance should be stored and
+re-used.
+
+:stop_sign: You should avoid invalid instrument names.
+
+> [!NOTE] OpenTelemetry will not collect metrics from instruments that are using
+> invalid names. Refer to the [OpenTelemetry
+  Specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-name-syntax)
+  for the valid syntax.
+
+:stop_sign: You should avoid changing the order of attributes while reporting
+measurements.
+
+> [!WARNING] The last line of code has bad performance since the attributes are
+> not following the same order as before:
+
+```rust
+let counter = meter.u64_counter("example_counter").build();
```

Comment:
Same as this: https://github.com/open-telemetry/opentelemetry-rust/pull/2946#discussion_r2067445782